server/data/data-jmap/src/main/java/org/apache/james/jmap/api/change/MailboxChangeRepository.java

```diff
@@ -31,5 +31,7 @@
 
     Mono<Void> save(MailboxChange change);
 
-    Mono<MailboxChanges> getSinceState(AccountId accountId, State state, Optional<Limit> maxIdsToReturn);
+    Mono<MailboxChanges> getSinceState(AccountId accountId, State state, Optional<Limit> maxChanges);
+
+    Mono<MailboxChanges> getSinceState(AccountId accountId, State state, Optional<Limit> maxChanges, boolean delegated);
```

Comment:
Boolean parameters in methods are generally a code smell.

How about:

```

    Mono<MailboxChanges> getSinceState(AccountId accountId, State state, Optional<Limit> maxChanges);

    Mono<MailboxChanges> getSinceStateWithDelegation(AccountId accountId, State state, Optional<Limit> maxChanges);
```

?